src/cryptsetup/cryptsetup-tokens/luks2-tpm2.c

```diff
@@ -50,6 +54,21 @@ int acquire_luks2_key(
         if ((flags & TPM2_FLAGS_USE_PIN) && !pin)
                 return -ENOANO;
 
+        /* salt xor pin, ie salt and pin must be both set or not set */
+        if (!!salt != !!pin)
+                return -ENOANO;
```

Comment:
> hmm, we currently don't salt our stuff. this check (unless i misunderstand this) would make sure that PIN enrollments with older systemd could not be unlocked anymore? or what i am missing?
> 
> if so, this needs to be weakened: use the salt if we have it, otherwise don't.

duh this is the unseal path, of course this check needs to be weakened.

Reply:
@poettering what about this:
```C
        /* If we're using a PIN, and the luks header has a salt, it better have a pin too */
        if ((flags & TPM2_FLAGS_USE_PIN) && salt && !pin)
                return -ENOANO;
```

Reply:
lgtm